test/Microsoft.IdentityModel.Tests/ReferenceXml.cs

```diff
@@ -561,6 +562,215 @@ public static KeyInfoTestSet KeyInfoMalformedCertificate
+        {
+            get
+            {
+                string[] audiences = new string[]
+                    {
+                        XmlGenerator.SamlAudienceXml(string.Empty)
+                    };
+
+                return new SamlAudienceRestrictionConditionTestSet
+                {
+                    SamlAudienceRestrictionCondition = Default.SamlAudienceRestrictionConditionSingleAudience,
+                    Xml = XmlGenerator.SamlAudienceRestrictionConditionXml(audiences)
+                };
+            }
+        }
+
+        public static SamlAudienceRestrictionConditionTestSet SamlAudienceRestrictionConditionInvaidElement
+        {
+            get
+            {
+                return new SamlAudienceRestrictionConditionTestSet
+                {
+                    SamlAudienceRestrictionCondition = Default.SamlAudienceRestrictionConditionSingleAudience,
+                    Xml = XmlGenerator.SamlAudienceRestrictionConditionXml(new string[] { XmlGenerator.SamlActionXml(null, null, null) })
+                };
+            }
+        }
+
+        public static SamlAudienceRestrictionConditionTestSet SamlAudienceRestrictionConditionSingleAudience
+        {
+            get
+            {
+                string[] audiences = new string[]
+                    {
+                        XmlGenerator.SamlAudienceXml(Default.Audience)
+                    };
+
+                return new SamlAudienceRestrictionConditionTestSet
+                {
+                    SamlAudienceRestrictionCondition = Default.SamlAudienceRestrictionConditionSingleAudience,
+                    Xml = XmlGenerator.SamlAudienceRestrictionConditionXml(audiences)
+                };
+            }
+        }
+
+        public static SamlAudienceRestrictionConditionTestSet SamlAudienceRestrictionConditionMultiAudience
+        {
+            get
+            {
+                List<string> audiences = new List<string>();
+                foreach (var audience in Default.Audiences)
+                {
+                    audiences.Add(XmlGenerator.SamlAudienceXml(audience));
+                }
+
+                return new SamlAudienceRestrictionConditionTestSet
+                {
+                    SamlAudienceRestrictionCondition = Default.SamlAudienceRestrictionConditionMultiAudience,
+                    Xml = XmlGenerator.SamlAudienceRestrictionConditionXml(audiences)
+                };
+            }
+        }
+        #endregion
+
+        #region SamlConditions
+        public static SamlConditionsTestSet SamlConditionsNoNbf
+        {
+            get
+            {
+                return new SamlConditionsTestSet
+                {
+                    Conditions = new SamlConditions(DateTimeUtil.GetMinValue(DateTimeKind.Utc), Default.NotOnOrAfter, new List<SamlCondition> { Default.SamlAudienceRestrictionConditionSingleAudience }),
```

Comment:
I think it is OK to add Default.SamlConditions, but I am not big fan with it since we don't use it so far.